tests/e2e/tidbcluster/serial.go

```diff
@@ -589,6 +589,229 @@ var _ = ginkgo.Describe("[Serial]", func() {
+				// confirm the tidb haven't been changed
+				changed, err := utilpod.PodsAreChanged(c, tidbPods)()
+				if err != nil {
+					log.Logf("ERROR: meet error during verify tidb pods, err:%v", err)
+					return true, nil
+				}
+				if changed {
+					return true, nil
+				}
+				log.Logf("confirm tidb pods haven't been changed this time")
+
+				return false, nil
+			})
+			framework.ExpectEqual(err, wait.ErrWaitTimeout, "expect pd/tikv/tidb have been changed for 5 minutes")
+			log.Logf("Upgrade TidbCluster 1 with label version=old and pod has been rolling updated.")
+
+			ginkgo.By("Deploy TidbCluster 2 with label version=new")
+			tc2Name := "tidbcluster2"
+			tc2 := fixture.GetTidbCluster(ns, tc2Name, utilimage.TiDBV4Prev)
+			tc2.Spec.PD.Replicas = 1
+			tc2.Spec.TiKV.Replicas = 1
+			tc2.Spec.TiDB.Replicas = 1
+			tc2.Labels = map[string]string{"version": "new"}
+
+			err = genericCli.Create(context.TODO(), tc2)
+			framework.ExpectNoError(err, "Expected TiDB cluster created")
+			log.Logf("Finished deploying TidbCluster 2 with label version=new.")
+
+			ginkgo.By("wait for 1min and check that no Pod created")
+			getPods = func(ls string) ([]v1.Pod, error) {
+				listOptions := metav1.ListOptions{
+					LabelSelector: ls,
+				}
+				podList, err := c.CoreV1().Pods(ns).List(listOptions)
+				if err != nil {
+					return nil, err
+				}
+				return podList.Items, nil
+			}
+			tidbPods, _ = getPods(labels.SelectorFromSet(label.New().Instance(tc2Name).TiDB().Labels()).String())
+			framework.ExpectEqual(len(tidbPods), 0, "tidbcluster2 no Pod created")
+			log.Logf("confirm Tidbcluster 2 no pod be created")
+
+			ginkgo.By("Deploy TiDB Operator 2 with --selector=version=new")
+			operatorVersion = "v1.1.11"
+			ocfg2 := &tests.OperatorConfig{
+				Namespace:       ns + "-canary-deploy",
+				ReleaseName:     "operator2",
+				Tag:             operatorVersion,
+				Image:           fmt.Sprintf("pingcap/tidb-operator:%s", operatorVersion),
+				ImagePullPolicy: v1.PullIfNotPresent,
+				Selector:        []string{"version=new"},
+				// Dependent this PR: https://github.com/pingcap/tidb-operator/pull/3819
+				// AppendReleaseSuffix: true,
+			}
+			oa.DeployOperatorOrDie(ocfg2)
+			log.Logf("Finished deploying TiDB Operator 2 with --selector=version=new")
+
+			ginkgo.By("Check TidbCluster 2 is ready")
+			err = oa.WaitForTidbClusterReady(tc2, 6*time.Minute, 5*time.Second)
+			framework.ExpectNoError(err, "Expected TiDB cluster2 ready")
+			log.Logf("confirm TidbCluster 2 is ready")
+
+			ginkgo.By("Delete the default TiDB Operator")
+			oa.CleanOperatorOrDie(ocfg)
+			log.Logf("Finished delete default Operator")
+
+			ginkgo.By("Upgrade TiDB version of TidbCluster 2")
+			err = controller.GuaranteedUpdate(genericCli, tc2, func() error {
+				tc2.Spec.Version = utilimage.TiDBV4
+				return nil
+			})
+			framework.ExpectNoError(err, "failed to update TidbCluster 2 to upgrade tidb version to %v", utilimage.TiDBV4)
```

Comment:
Wait for cluster ready after the upgrade, for example:
```
			err = oa.WaitForTidbClusterReady(tc, 10*time.Minute, 10*time.Second)
			framework.ExpectNoError(err, "failed to wait for TidbCluster %s/%s components ready", ns, tc.Name)
```
Refer to https://github.com/pingcap/tidb-operator/pull/3775/files.

Reply:
add this:
```golang
			err = oa.WaitForTidbClusterReady(tc, 10*time.Minute, 10*time.Second)
			framework.ExpectNoError(err, "failed to wait for TidbCluster %s/%s components ready", ns, tc2.Name)

```